python/cudf/cudf/_libxx/aggregation.pyx

```diff
@@ -8,54 +10,171 @@ from libcpp.memory cimport unique_ptr
+from cudf._libxx.types cimport (
+    underlying_type_t_interpolation,
+    underlying_type_t_include_nulls
+)
+from cudf._libxx.types import Interpolation
 
 
-# need to update as and when we add new aggregations with additional options
-cdef unique_ptr[aggregation] get_aggregation(op, kwargs) except *:
-
-    cdef type_id tid
-    cdef data_type out_dtype
-    cdef string cpp_str
-    cdef unique_ptr[aggregation] agg
-
-    if op == "sum":
-        agg = move(libcudf_aggregation.make_sum_aggregation())
-    elif op == "min":
-        agg = move(libcudf_aggregation.make_min_aggregation())
-    elif op == "max":
-        agg = move(libcudf_aggregation.make_max_aggregation())
-    elif op == "mean":
-        agg = move(libcudf_aggregation.make_mean_aggregation())
-    elif op == "count":
-        agg = move(libcudf_aggregation.make_count_aggregation())
-    elif op == "any":
-        agg = move(libcudf_aggregation.make_any_aggregation())
-    elif op == "all":
-        agg = move(libcudf_aggregation.make_all_aggregation())
-    elif op == "product":
-        agg = move(libcudf_aggregation.make_product_aggregation())
-    elif op == "sum_of_squares":
-        agg = move(libcudf_aggregation.make_sum_of_squares_aggregation())
-    elif op == "var":
-        agg = move(libcudf_aggregation.make_variance_aggregation(
-            kwargs['ddof'])
-        )
-    elif op == "std":
-        agg = move(libcudf_aggregation.make_std_aggregation(kwargs['ddof']))
-    elif op == "count":
-        agg = move(libcudf_aggregation.make_count_aggregation())
+class AggregationKind(Enum):
+    SUM = libcudf_aggregation.aggregation.Kind.SUM
+    PRODUCT = libcudf_aggregation.aggregation.Kind.PRODUCT
+    MIN = libcudf_aggregation.aggregation.Kind.MIN
+    MAX = libcudf_aggregation.aggregation.Kind.MAX
+    COUNT = libcudf_aggregation.aggregation.Kind.COUNT_VALID
+    SIZE = libcudf_aggregation.aggregation.Kind.COUNT_ALL
+    ANY = libcudf_aggregation.aggregation.Kind.ANY
+    ALL = libcudf_aggregation.aggregation.Kind.ALL
+    SUM_OF_SQUARES = libcudf_aggregation.aggregation.Kind.SUM_OF_SQUARES
+    MEAN = libcudf_aggregation.aggregation.Kind.MEAN
+    VARIANCE = libcudf_aggregation.aggregation.Kind.VARIANCE
+    STD = libcudf_aggregation.aggregation.Kind.STD
+    MEDIAN = libcudf_aggregation.aggregation.Kind.MEDIAN
+    QUANTILE = libcudf_aggregation.aggregation.Kind.QUANTILE
+    ARGMAX = libcudf_aggregation.aggregation.Kind.ARGMAX
+    ARGMIN = libcudf_aggregation.aggregation.Kind.ARGMIN
+    NUNIQUE = libcudf_aggregation.aggregation.Kind.NUNIQUE
+    NTH_ELEMENT = libcudf_aggregation.aggregation.Kind.NTH_ELEMENT
+    PTX = libcudf_aggregation.aggregation.Kind.PTX
+    CUDA = libcudf_aggregation.aggregation.Kind.CUDA
+
+
+cdef class Aggregation:
+
+    def __init__(self, op, **kwargs):
+        self.c_obj = move(make_aggregation(op, kwargs))
+
+    @property
+    def kind(self):
+        return AggregationKind(self.c_obj.get()[0].kind).name.lower()
```

Comment:
My thought is that we keep the Python `Enum` as an internal detail, and have `kind` always return a `str`, since that's how users will typically specify an aggregation.